sway/ipc-server.c

```diff
@@ -226,7 +232,12 @@ int ipc_client_handle_readable(int client_fd, uint32_t mask, void *data) {
 	// Wait for the rest of the command payload in case the header has already been read
 	if (client->payload_length > 0) {
 		if ((uint32_t)read_available >= client->payload_length) {
-			ipc_client_handle_command(client);
+			// Pass the payload_length and payload_type so that
+			// they are semantically and logically decoupled from
+			// the logic pertaining to the values stored in client.
+			ipc_client_handle_command(client,
```

Comment:
Unless it's inaccurate, then I'm not sure what you mean by that.

Reply:
I just don't think the comment adds anything of value. The comment in the struct declaration should be sufficient

Reply:
> I wanted to explain to people in the future why I decided to pass in the arguments rather than the way it was before. Unless you're referring to the duplication of the comment, but I would rather be verbose than the opposite.

Can you try rewording it in a way that provides some reasoning then? If I just started looking at the code and came across that comment, I would question why the property is part of the struct.

Maybe the properties should also be renamed to something like `pending_length` and `pending_type`?